### PR TITLE
View compilation

### PR DESCRIPTION
**Pull request by Rick-Anderson**

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@tdykstra  please review

**Comment by Rick-Anderson** [comment]
Fixes #2307

**Comment by Rick-Anderson** [comment]
@danroth27 Should I add a link to view compilation in "additional resources" in the publishing docs?

**Comment by tdykstra** [comment]
Or in performance?

**Comment by Rick-Anderson** [comment]
Or both?

File: aspnetcore/mvc/views/razor.md [hunk 1]
```diff
@@ -567,6 +567,20 @@ C# Razor keywords need to be double escaped with `@(@C# Razor Keyword)`, for exa
 * namespace
 * class
 
+## View compilation
+
+Razor views are compiled at runtime when the view is invoked. If you prefer to compile your Razor views and deploy them with your app follow these steps:
+
+1. Add a reference to "Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design" under the "dependencies" section.
+2. Add a reference to "Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Tools" under the "tools" section.
+3. Add a postpublish script to invoke view compiler:
```
Comment (tdykstra) [inline aspnetcore/mvc/views/razor.md]: invoke **the** view compiler

File: aspnetcore/mvc/views/razor.md [hunk 2]
```diff
@@ -567,6 +567,20 @@ C# Razor keywords need to be double escaped with `@(@C# Razor Keyword)`, for exa
...
+
+```
```
Comment (tdykstra) [inline aspnetcore/mvc/views/razor.md]: indent 2 to put it under step 3

Comment (Rick-Anderson) [inline aspnetcore/mvc/views/razor.md]: @tdykstra  not sure how to write it more neutral and make it work at the same time. We'll have to revisit this one.

Comment (tdykstra) [inline aspnetcore/mvc/views/razor.md]: Maybe add the word "project.json" somewhere to ensure it turns up in searches when we look for affected articles.

**Comment by Rick-Anderson** [comment]
@tdykstra  added project.json

**Pull request merged**